### PR TITLE
feat: added SARA as a provider and respective S3 product definitions.

### PR DESCRIPTION
**Pull request by catchSheep**
### Your checklist for this pull request
Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
    - As I've synced quite a few times without rebasing (whoops), it may be better to squash merge these changes.
- [x] Check your code additions will fail neither code linting checks nor unit test.
    - Currently passes all tests  the original passes except "tests/test_cli.py::TestEodagCli::test_eodag_cli_version", which may possibly just be on my end

### Description
Fixes/adds support for issues https://github.com/CS-SI/eodag/issues/575, https://github.com/CS-SI/eodag/issues/168 and https://github.com/CS-SI/eodag/issues/14. Adding  provider [SARA](https://copernicus.nci.org.au/sara.client/#/home) and SARA product types (eg S3 water products, SLSTR AOD & FRP, and synergy products).
The changes should only affect eodag/resources/providers.yml,  eodag/resources/product_types.yml, and update the test dictionaries with the newly added product types.

Also fixes  https://github.com/CS-SI/eodag/issues/577

### CODE REVIEW

**Review by sbrunato — CHANGES_REQUESTED** [review]

**Review by sbrunato — CHANGES_REQUESTED** [review]

**Review by sbrunato — CHANGES_REQUESTED** [review]
Hello @catchSheep , I wish you all the best for 2023! Sorry for the delay in reviewing this PR. 

Could you please rebase it on up-to-date `develop` branch in order to include #580 that fixes the failing test in github actions ?
Thanks

**Comment by catchSheep** [comment]
> Hello @catchSheep , I wish you all the best for 2023! Sorry for the delay in reviewing this PR.
>
> Could you please rebase (or merge) it on up-to-date develop branch in order to include https://github.com/CS-SI/eodag/pull/580 that fixes the failing test in github actions ?
Thanks

No worries, happy new year to you too.

I had some issues merging the updates, so I re-wrote the history to the current CS-SI eodag develop branch, and git cherry-picked the relevant changes back in. It seems to be passing all relevant checks for python 3.9 & 3.10

File: eodag/resources/providers.yml [hunk 1]
```diff
@@ -1016,6 +1016,10 @@
       generic_product_type_id: '$.id'
       generic_product_type_parsable_properties:
         collection: '$.id'
+      generic_product_type_unparsable_properties:
+        # cloud cover search disabled for unknown product_type
+        metadata_mapping:
+          cloudCover: '$.null'
```
Comment (sbrunato) [inline eodag/resources/providers.yml:1022]:
```suggestion
```
restores https://github.com/CS-SI/eodag/pull/530

File: eodag/resources/providers.yml [hunk 2]
```diff
@@ -1254,7 +1258,7 @@
   url: https://mundiwebservices.com/
   search: !plugin
     type: QueryStringSearch
-    api_endpoint: 'https://{collection}.browse.catalog.mundiwebservices.com/opensearch'
+    api_endpoint: 'https://mundiwebservices.com/acdc/catalog/proxy/search/{collection}/opensearch'
```
Comment (sbrunato) [inline eodag/resources/providers.yml:1261]:
```suggestion
    api_endpoint: 'https://{collection}.browse.catalog.mundiwebservices.com/opensearch'
```
restores https://github.com/CS-SI/eodag/pull/524

File: eodag/resources/providers.yml [hunk 3]
```diff
@@ -1383,7 +1387,6 @@
     base_uri: 'https://mundiwebservices.com/dp'
     extract: true
     auth_error_code: 401
```
Comment (sbrunato) [inline eodag/resources/providers.yml:1389]:
```suggestion
    auth_error_code: 401
    bucket_path_level: 0
```
restores https://github.com/CS-SI/eodag/pull/524

**Review by sbrunato — CHANGES_REQUESTED** [review]

**Review by sbrunato — APPROVED** [review]

**Pull request merged**